src/Compilers/Core/Portable/InternalUtilities/FatalError.cs

```diff
@@ -225,7 +225,7 @@ public static bool ReportAndCatchUnlessCanceled(Exception exception, Cancellatio
 
 #endif
 
-        private static readonly object s_reportedMarker = new();
+        private static readonly Guid s_reportedMarker = Guid.NewGuid();
```

Comment:
Might as well box this so we don't repeatedly allocate the box.

```suggestion
        private static readonly object s_reportedMarker = Guid.NewGuid();
```

Reply:
Note that if you make this change, you'll also need to change the `==` operators to use `Equals` instead. They'll be just as fast for the identity cases, but when it crosses an RPC boundary it will switch to use value equality. The other option is to check it for null instead.

Reply:
Thanks. The only two places this is checked is already doing a null check, so should be good.